vavr/src-gen/main/java/io/vavr/Tuple1.java

```diff
@@ -142,6 +141,73 @@ public T1 _1() {
         return List.of(_1);
     }
 
+    public <T2> Tuple2<T1, T2> concat(T2 t2) {
```

Comment:
It can be ambiguous with `concat(Tuple1<T2> tuple)` (in case if `T2` not shown intentionally). What do you think about rename this method to `add(T2)` and keep others as `concat(Tuple*<***>)`?

Reply:
Yes, we had problems in the past with overloads that have a generic value parameter. However, my tests do not lead to ambiguities. Unless we find a counter-example, I would keep same names.

```java
class Tuple2<T1, T2> {

    // ...

    public final <T3> Tuple3<T1, T2, T3> concat(T3 value) {
        return new Tuple3<>(_1, _2, value);
    }

    public final <T3> Tuple3<T1, T2, T3> concat(Tuple1<T3> tuple) {
        return new Tuple3<>(_1, _2, tuple._1);
    }

}

// both tests compile fine
final Tuple3<String, Boolean, Integer> testee1 = Tuple.of("a", true).concat(1);
final Tuple3<String, Boolean, Integer> testee2 = Tuple.of("a", true).concat(Tuple.of(1));
```

Reply:
```java
// fail
Tuple3<String, Boolean, Integer> testee3 =
        Tuple.of("a", true).concat(Tuple.of(1)).map3(third -> third._1);
// compiles fine
Tuple3<String, Boolean, Integer> testee4 = 
        Tuple.of("a", true).<Tuple1<Integer>>concat(Tuple.of(1)).map3(third -> third._1);
```

Reply:
Yes, Ruslan is right. In the case of values T1, ... Tn we should choose another name. I would prefer **`append`** over `add` because our sequential types also have `append` (in contrast to Set which has `add`).